### DISCUSSION TITLE
How the range request actually work?

### DISCUSSION THREAD

**Discussion by xingya1822** [mozilla/pdf.js#18609] [Q&A]
> https://github.com/mozilla/pdf.js/wiki/Frequently-Asked-Questions#range

according to this FAQ, 
Why the very first HTTP request made to server seems loaded the whole file ?
![图片](https://github.com/user-attachments/assets/7983fdd4-1b0c-4764-ae99-6f9b0d6ef101)

And why all those following requests  were made ? While I didn't begin to scroll down the pdf ?

**Comment by jiangxiaoqiang** [comment]
> > https://github.com/mozilla/pdf.js/wiki/Frequently-Asked-Questions#range
> 
> according to this FAQ, Why the very first HTTP request made to server seems loaded the whole file ? ![图片](https://private-user-images.githubusercontent.com/175574802/358149588-7983fdd4-1b0c-4764-ae99-6f9b0d6ef101.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU3OTcxNjUsIm5iZiI6MTcyNTc5Njg2NSwicGF0aCI6Ii8xNzU1NzQ4MDIvMzU4MTQ5NTg4LTc5ODNmZGQ0LTFiMGMtNDc2NC1hZTk5LTZmOWIwZDZlZjEwMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkwOFQxMjAxMDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wNzE1NWVjMTBhOTY1ODc3YzlmOTNkMDVhYmM5ZWI4MGIwMTkyMmI2N2JhY2UwNjVmZGE1NGQwOTczNDNkNWExJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.aevE128syNA7xdEknOORISzYcbIzB1sjMmNWjiDLEZk)
> 
> And why all those following requests were made ? While I didn't begin to scroll down the pdf ?

I have the same confusing. did you figure out this issue?

**Comment by SamWiIIiams** [comment]
Notice that the response code is 200 for that first call. All of the other responses should be 206. The first call to the server requests the entire document and returns the metadata (contentLength) needed to issue the subsequent range requests. By default - in my experience at least - it will request the entire document and the chunks necessary to display the pages that are on screen. The initial request will continue streaming the entire document in the backend regardless of how many chunks you have requested. If you add `disableStream` to the request, it will no longer do this and instead only load the document in chunks.

**Reply by SamWiIIiams** [reply]
Additionally, if you don't want it to continue to load chunks you can use `disableAutoFetch`
